src/shared/settings.ts

```diff
@@ -255,8 +256,17 @@ function createSettingsClass<T extends TypeDescriptor>(section: string, descript
 
                 for (const key of props.filter(isDifferent)) {
                     this.log(`key "${key}" changed`)
-                    store[key] = this.get(key)
-                    emitter.fire({ key })
+                    // getChangedEmitter() can't provide a default value to
+                    // this.get(), so we need to catch errors here.
+                    try {
+                        store[key] = this.get(key)
+                        emitter.fire({ key })
+                    } catch (e) {
+                        if (this.throwInvalid) {
+                            throw e
```

Comment:
`getChangedEmitter` also can't know a default value to provide, and also we probably don't want async events throwing randomly. So we let get() log the failure and skip the emitter.